### PR TITLE
Simplify gdal_baseline_slope.py

### PR DESCRIPTION
**Pull request by dpmayerUSGS**
I have simplified gdal_baseline_slope.py by condensing the original three baseline slope functions into a single function that can accept any pixel baseline value (as long as it's less than half the smallest image dimension). 

The revised baseline slope function also uses the "footprint" keyword for ndimage.generic_filter, rather than "size." This eliminates the need to create (potentially large) arrays in memory that contain dummy values not used in the actual slope calculations.

### CODE REVIEW

File: gdal_baseline_slope/gdal_baseline_slope.py [hunk 1]
```diff
@@ -284,29 +254,18 @@ def calc_slope(in_filter, x_cellsize, y_cellsize, noData):
...
-   if baseline == 1:
-      slope = generic_filter(raster_data, calc_slope_baseline1, size=2, mode='constant',
-                       cval=outNoData, extra_arguments=(cellsizeX, cellsizeY, outNoData))
-      #shift half a pixel to center 
-      newGeomatrix = (X1 , geomatrix[1], geomatrix[2], Y1, geomatrix[4], geomatrix[5])
+   
+   if baseline is not None:
+      slope = generic_filter(raster_data, calc_slope_baseline, footprint=footprint, mode='constant',
+                             cval=outNoData, extra_arguments=(cellsizeX, cellsizeY, baseline, outNoData))
+      ## If baseline is an even number, the new geomatrix will be shifted half a pixel,
```
Comment (thareUSGS) [inline gdal_baseline_slope/gdal_baseline_slope.py]: suppose to read "odd" number right?

File: gdal_baseline_slope/gdal_baseline_slope.py [hunk 2]
```diff
@@ -4,8 +4,9 @@
 #  Name:     gdal_baseline_slope.py 
 #  Project:  GDAL Python Interface
 #  Purpose:  Given a DEM, calculate specialized slopes using various baseline
-#            lengths (1 baseline, 2 baseline, 5 baseline) as defined by R. Kirk. 
-#            The more commonly-used Horn's Method is also available.
+#            lengths (measured in pixels) as defined by R. Kirk. 
+#            If baseline is not specified, slopes will be calculated using
+#            Horn's Method.
 #  Author:   Trent Hare, thare@usgs.gov
```
Comment (thareUSGS) [inline gdal_baseline_slope/gdal_baseline_slope.py]: I know history in github, but good to add your name to history.

Comment (thareUSGS) [inline gdal_baseline_slope/gdal_baseline_slope.py]: I know history in github, but good to add your name to history.

File: gdal_baseline_slope/gdal_baseline_slope.py [hunk 3]
```diff
@@ -53,7 +54,7 @@
 # Usage()
 def Usage():
     print("""
-Usage: gdal_baseline_slope.py [-baseline 1,2,5] [-ot Byte] [-crop]  infile outfile.tif
+    Usage: gdal_baseline_slope.py [-baseline [integer]] [-ot Byte] [-crop]  infile outfile.tif
```
Comment (thareUSGS) [inline gdal_baseline_slope/gdal_baseline_slope.py]: Probably need to add a comment in usage or description on intended "integers" - e.g. 1, 2, 5. I should add more documentation for used with LROC, HiRISE, CTX and more details.

File: gdal_baseline_slope/gdal_baseline_slope.py (hunk 1 above)
Comment (dpmayerUSGS) [reply gdal_baseline_slope/gdal_baseline_slope.py]: Yes, that should read "odd."
Good catch.

File: gdal_baseline_slope/gdal_baseline_slope.py (hunk 3 above)
Comment (dpmayerUSGS) [reply gdal_baseline_slope/gdal_baseline_slope.py]: Not a bad idea. Maybe open as a separate issue if you decide to merge this PR?
I was trying to fulfill the "TODO" listed in the earlier version that mused about adding support for arbitrary baselines.

File: gdal_baseline_slope/gdal_baseline_slope.py (hunk 2 above)
Comment (dpmayerUSGS) [reply gdal_baseline_slope/gdal_baseline_slope.py]: Sure. I'm never quite sure how to handle author credit on these kinds of things. 
Does that need to be part of this PR or can that just be a small commit later?

File: gdal_baseline_slope/gdal_baseline_slope.py [hunk 4]
```diff
@@ -363,16 +322,16 @@ def calc_slope(in_filter, x_cellsize, y_cellsize, noData):
          iBand8 = outdataset8.GetRasterBand(band)
          cropBand8 = cropdataset8.GetRasterBand(band)
       if baseline == 1:
```
Comment (thareUSGS) [inline gdal_baseline_slope/gdal_baseline_slope.py]: TODO: currently only cropping for 1,2,5. should this be also made generic to crop based on input baseline integer?

**Pull request merged**